presto-client/src/test/java/com/facebook/presto/client/TestQueryResults.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.client;
+
+import com.facebook.presto.spi.type.NamedTypeSignature;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.TypeSignatureParameter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Base64;
+import java.util.List;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.CharType.createCharType;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Lists.newArrayList;
+import static org.testng.Assert.assertEquals;
+
+public class TestQueryResults
+{
+    @Test
+    public void testFixData()
+    {
+        assertQueryResult(new Column("a", StandardTypes.BIGINT, new ClientTypeSignature(BIGINT.getTypeSignature())), 1000, 1000L);
```

Comment:
The way Column is constructed here is consistent with how it looks in reality. It preserves the same behavior where the string type (second parameter) and the structured type (third parameter) are different as currently are in production. I will change this to be simpler, using TypeSignature to construct Column. The test will still work. But Column will look different from what they actually are in production. It would arguably make the test less accurate, though this will be a hard to maintain thing anyways. How column is constructed now (the version in production) is not exactly logically consistent (as we discussed earlier offline), that's why things look a bit awkward. And it's not the scope of this PR to fix those.